### PR TITLE
python3Packages.djangorestframework: 3.14.0 -> 3.15.1

### PR DESCRIPTION
**Pull request by sephii**
## Description of changes

https://github.com/encode/django-rest-framework/releases/tag/3.15.1

Required by https://github.com/NixOS/nixpkgs/pull/309466

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/djangorestframework/default.nix
Comment (matthiasbeyer) [inline pkgs/development/python-modules/djangorestframework/default.nix:14]: This diff has nothing to do with the version update commit, please split it to another commit.

**Review by matthiasbeyer — DISMISSED** [review]

**Comment by matthiasbeyer** [comment]
Result of `nixpkgs-review pr 315187` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>14 packages marked as broken and skipped:</summary>
  <ul>
    <li>baserow</li>
    <li>baserow.dist</li>
    <li>netbox_3_6</li>
    <li>python311Packages.djangorestframework-guardian</li>
    <li>python311Packages.djangorestframework-guardian.dist</li>
    <li>python312Packages.django-countries</li>
    <li>python312Packages.django-countries.dist</li>
    <li>python312Packages.django-graphiql-debug-toolbar</li>
    <li>python312Packages.django-graphiql-debug-toolbar.dist</li>
    <li>python312Packages.djangorestframework-guardian</li>
    <li>python312Packages.djangorestframework-guardian.dist</li>
    <li>python312Packages.graphene-django</li>
    <li>python312Packages.graphene-django.dist</li>
    <li>seahub</li>
  </ul>
</details>
<details>
  <summary>9 packages failed to build:</summary>
  <ul>
    <li>paperless-ngx</li>
    <li>pretalx</li>
    <li>pretalx.dist</li>
    <li>pretalx.static</li>
    <li>pretix</li>
    <li>pretix.dist</li>
    <li>python312Packages.django-dynamic-preferences</li>
    <li>python312Packages.django-dynamic-preferences.dist</li>
    <li>tandoor-recipes</li>
  </ul>
</details>
<details>
  <summary>165 packages built:</summary>
  <ul>
    <li>authentik</li>
    <li>mailman-web</li>
    <li>mailman-web.dist</li>
    <li>netbox (netbox_3_7)</li>
    <li>peering-manager</li>
    <li>python311Packages.deal</li>
    <li>python311Packages.deal.dist</li>
    <li>python311Packages.dj-rest-auth</li>
    <li>python311Packages.dj-rest-auth.dist</li>
    <li>python311Packages.django-celery-beat</li>
    <li>python311Packages.django-celery-beat.dist</li>
    <li>python311Packages.django-countries</li>
    <li>python311Packages.django-countries.dist</li>
    <li>python311Packages.django-dynamic-preferences</li>
    <li>python311Packages.django-dynamic-preferences.dist</li>
    <li>python311Packages.django-filter</li>
    <li>python311Packages.django-filter.dist</li>
    <li>python311Packages.django-graphiql-debug-toolbar</li>
    <li>python311Packages.django-graphiql-debug-toolbar.dist</li>
    <li>python311Packages.django-i18nfield</li>
    <li>python311Packages.django-i18nfield.dist</li>
    <li>python311Packages.django-login-required-middleware</li>
    <li>python311Packages.django-login-required-middleware.dist</li>
    <li>python311Packages.django-modelcluster</li>
    <li>python311Packages.django-modelcluster.dist</li>
    <li>python311Packages.django-oauth-toolkit</li>
    <li>python311Packages.django-oauth-toolkit.dist</li>
    <li>python311Packages.django-payments</li>
    <li>python311Packages.django-payments.dist</li>
    <li>python311Packages.django-phonenumber-field</li>
    <li>python311Packages.django-phonenumber-field.dist</li>
    <li>python311Packages.django-postgresql-netfields</li>
    <li>python311Packages.django-postgresql-netfields.dist</li>
    <li>python311Packages.django-rest-auth</li>
    <li>python311Packages.django-rest-auth.dist</li>
    <li>python311Packages.django-rest-polymorphic</li>
    <li>python311Packages.django-rest-polymorphic.dist</li>
    <li>python311Packages.django-rest-registration</li>
    <li>python311Packages.django-rest-registration.dist</li>
    <li>python311Packages.django-taggit</li>
    <li>python311Packages.django-taggit.dist</li>
    <li>python311Packages.django-timezone-field</li>
    <li>python311Packages.django-timezone-field.dist</li>
    <li>python311Packages.django-two-factor-auth</li>
    <li>python311Packages.django-two-factor-auth.dist</li>
    <li>python311Packages.djangorestframework</li>
    <li>python311Packages.djangorestframework-camel-case</li>
    <li>python311Packages.djangorestframework-camel-case.dist</li>
    <li>python311Packages.djangorestframework-dataclasses</li>
    <li>python311Packages.djangorestframework-dataclasses.dist</li>
    <li>python311Packages.djangorestframework-guardian2</li>
    <li>python311Packages.djangorestframework-guardian2.dist</li>
    <li>python311Packages.djangorestframework-recursive</li>
    <li>python311Packages.djangorestframework-recursive.dist</li>
    <li>python311Packages.djangorestframework-simplejwt</li>
    <li>python311Packages.djangorestframework-simplejwt.dist</li>
    <li>python311Packages.djangorestframework.dist</li>
    <li>python311Packages.drf-jwt</li>
    <li>python311Packages.drf-jwt.dist</li>
    <li>python311Packages.drf-nested-routers</li>
    <li>python311Packages.drf-nested-routers.dist</li>
    <li>python311Packages.drf-spectacular</li>
    <li>python311Packages.drf-spectacular.dist</li>
    <li>python311Packages.drf-ujson2</li>
    <li>python311Packages.drf-ujson2.dist</li>
    <li>python311Packages.drf-writable-nested</li>
    <li>python311Packages.drf-writable-nested.dist</li>
    <li>python311Packages.drf-yasg</li>
    <li>python311Packages.drf-yasg.dist</li>
    <li>python311Packages.graphene-django</li>
    <li>python311Packages.graphene-django.dist</li>
    <li>python311Packages.icontract</li>
    <li>python311Packages.icontract.dist</li>
    <li>python311Packages.permissionedforms</li>
    <li>python311Packages.permissionedforms.dist</li>
    <li>python311Packages.pylddwrap</li>
    <li>python311Packages.pylddwrap.dist</li>
    <li>python311Packages.rules</li>
    <li>python311Packages.rules.dist</li>
    <li>python311Packages.vaa</li>
    <li>python311Packages.vaa.dist</li>
    <li>python311Packages.wagtail</li>
    <li>python311Packages.wagtail-factories</li>
    <li>python311Packages.wagtail-factories.dist</li>
    <li>python311Packages.wagtail-localize</li>
    <li>python311Packages.wagtail-localize.dist</li>
    <li>python311Packages.wagtail-modeladmin</li>
    <li>python311Packages.wagtail-modeladmin.dist</li>
    <li>python311Packages.wagtail.dist</li>
    <li>python312Packages.deal</li>
    <li>python312Packages.deal.dist</li>
    <li>python312Packages.dj-rest-auth</li>
    <li>python312Packages.dj-rest-auth.dist</li>
    <li>python312Packages.django-celery-beat</li>
    <li>python312Packages.django-celery-beat.dist</li>
    <li>python312Packages.django-filter</li>
    <li>python312Packages.django-filter.dist</li>
    <li>python312Packages.django-i18nfield</li>
    <li>python312Packages.django-i18nfield.dist</li>
    <li>python312Packages.django-login-required-middleware</li>
    <li>python312Packages.django-login-required-middleware.dist</li>
    <li>python312Packages.django-modelcluster</li>
    <li>python312Packages.django-modelcluster.dist</li>
    <li>python312Packages.django-oauth-toolkit</li>
    <li>python312Packages.django-oauth-toolkit.dist</li>
    <li>python312Packages.django-payments</li>
    <li>python312Packages.django-payments.dist</li>
    <li>python312Packages.django-phonenumber-field</li>
    <li>python312Packages.django-phonenumber-field.dist</li>
    <li>python312Packages.django-postgresql-netfields</li>
    <li>python312Packages.django-postgresql-netfields.dist</li>
    <li>python312Packages.django-rest-auth</li>
    <li>python312Packages.django-rest-auth.dist</li>
    <li>python312Packages.django-rest-polymorphic</li>
    <li>python312Packages.django-rest-polymorphic.dist</li>
    <li>python312Packages.django-rest-registration</li>
    <li>python312Packages.django-rest-registration.dist</li>
    <li>python312Packages.django-taggit</li>
    <li>python312Packages.django-taggit.dist</li>
    <li>python312Packages.django-timezone-field</li>
    <li>python312Packages.django-timezone-field.dist</li>
    <li>python312Packages.django-two-factor-auth</li>
    <li>python312Packages.django-two-factor-auth.dist</li>
    <li>python312Packages.djangorestframework</li>
    <li>python312Packages.djangorestframework-camel-case</li>
    <li>python312Packages.djangorestframework-camel-case.dist</li>
    <li>python312Packages.djangorestframework-dataclasses</li>
    <li>python312Packages.djangorestframework-dataclasses.dist</li>
    <li>python312Packages.djangorestframework-guardian2</li>
    <li>python312Packages.djangorestframework-guardian2.dist</li>
    <li>python312Packages.djangorestframework-recursive</li>
    <li>python312Packages.djangorestframework-recursive.dist</li>
    <li>python312Packages.djangorestframework-simplejwt</li>
    <li>python312Packages.djangorestframework-simplejwt.dist</li>
    <li>python312Packages.djangorestframework.dist</li>
    <li>python312Packages.drf-jwt</li>
    <li>python312Packages.drf-jwt.dist</li>
    <li>python312Packages.drf-nested-routers</li>
    <li>python312Packages.drf-nested-routers.dist</li>
    <li>python312Packages.drf-spectacular</li>
    <li>python312Packages.drf-spectacular.dist</li>
    <li>python312Packages.drf-ujson2</li>
    <li>python312Packages.drf-ujson2.dist</li>
    <li>python312Packages.drf-writable-nested</li>
    <li>python312Packages.drf-writable-nested.dist</li>
    <li>python312Packages.drf-yasg</li>
    <li>python312Packages.drf-yasg.dist</li>
    <li>python312Packages.icontract</li>
    <li>python312Packages.icontract.dist</li>
    <li>python312Packages.permissionedforms</li>
    <li>python312Packages.permissionedforms.dist</li>
    <li>python312Packages.pylddwrap</li>
    <li>python312Packages.pylddwrap.dist</li>
    <li>python312Packages.rules</li>
    <li>python312Packages.rules.dist</li>
    <li>python312Packages.vaa</li>
    <li>python312Packages.vaa.dist</li>
    <li>python312Packages.wagtail</li>
    <li>python312Packages.wagtail-factories</li>
    <li>python312Packages.wagtail-factories.dist</li>
    <li>python312Packages.wagtail-localize</li>
    <li>python312Packages.wagtail-localize.dist</li>
    <li>python312Packages.wagtail-modeladmin</li>
    <li>python312Packages.wagtail-modeladmin.dist</li>
    <li>python312Packages.wagtail.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/djangorestframework/default.nix
Comment (mweinelt) [reply pkgs/development/python-modules/djangorestframework/default.nix]: We recently reformatted all packages with `nixfmt-rfc-style`, and we'd very much like to keep that style to prevent noise caused by reformatting.

**Comment by matthiasbeyer** [comment]
Result of `nixpkgs-review pr 315187` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>14 packages marked as broken and skipped:</summary>
  <ul>
    <li>baserow</li>
    <li>baserow.dist</li>
    <li>netbox_3_6</li>
    <li>python311Packages.djangorestframework-guardian</li>
    <li>python311Packages.djangorestframework-guardian.dist</li>
    <li>python312Packages.django-countries</li>
    <li>python312Packages.django-countries.dist</li>
    <li>python312Packages.django-graphiql-debug-toolbar</li>
    <li>python312Packages.django-graphiql-debug-toolbar.dist</li>
    <li>python312Packages.djangorestframework-guardian</li>
    <li>python312Packages.djangorestframework-guardian.dist</li>
    <li>python312Packages.graphene-django</li>
    <li>python312Packages.graphene-django.dist</li>
    <li>seahub</li>
  </ul>
</details>
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>python312Packages.django-dynamic-preferences</li>
    <li>python312Packages.django-dynamic-preferences.dist</li>
  </ul>
</details>
<details>
  <summary>172 packages built:</summary>
  <ul>
    <li>authentik</li>
    <li>mailman-web</li>
    <li>mailman-web.dist</li>
    <li>netbox (netbox_3_7)</li>
    <li>paperless-ngx</li>
    <li>peering-manager</li>
    <li>pretalx</li>
    <li>pretalx.dist</li>
    <li>pretalx.static</li>
    <li>pretix</li>
    <li>pretix.dist</li>
    <li>python311Packages.deal</li>
    <li>python311Packages.deal.dist</li>
    <li>python311Packages.dj-rest-auth</li>
    <li>python311Packages.dj-rest-auth.dist</li>
    <li>python311Packages.django-celery-beat</li>
    <li>python311Packages.django-celery-beat.dist</li>
    <li>python311Packages.django-countries</li>
    <li>python311Packages.django-countries.dist</li>
    <li>python311Packages.django-dynamic-preferences</li>
    <li>python311Packages.django-dynamic-preferences.dist</li>
    <li>python311Packages.django-filter</li>
    <li>python311Packages.django-filter.dist</li>
    <li>python311Packages.django-graphiql-debug-toolbar</li>
    <li>python311Packages.django-graphiql-debug-toolbar.dist</li>
    <li>python311Packages.django-i18nfield</li>
    <li>python311Packages.django-i18nfield.dist</li>
    <li>python311Packages.django-login-required-middleware</li>
    <li>python311Packages.django-login-required-middleware.dist</li>
    <li>python311Packages.django-modelcluster</li>
    <li>python311Packages.django-modelcluster.dist</li>
    <li>python311Packages.django-oauth-toolkit</li>
    <li>python311Packages.django-oauth-toolkit.dist</li>
    <li>python311Packages.django-payments</li>
    <li>python311Packages.django-payments.dist</li>
    <li>python311Packages.django-phonenumber-field</li>
    <li>python311Packages.django-phonenumber-field.dist</li>
    <li>python311Packages.django-postgresql-netfields</li>
    <li>python311Packages.django-postgresql-netfields.dist</li>
    <li>python311Packages.django-rest-auth</li>
    <li>python311Packages.django-rest-auth.dist</li>
    <li>python311Packages.django-rest-polymorphic</li>
    <li>python311Packages.django-rest-polymorphic.dist</li>
    <li>python311Packages.django-rest-registration</li>
    <li>python311Packages.django-rest-registration.dist</li>
    <li>python311Packages.django-taggit</li>
    <li>python311Packages.django-taggit.dist</li>
    <li>python311Packages.django-timezone-field</li>
    <li>python311Packages.django-timezone-field.dist</li>
    <li>python311Packages.django-two-factor-auth</li>
    <li>python311Packages.django-two-factor-auth.dist</li>
    <li>python311Packages.djangorestframework</li>
    <li>python311Packages.djangorestframework-camel-case</li>
    <li>python311Packages.djangorestframework-camel-case.dist</li>
    <li>python311Packages.djangorestframework-dataclasses</li>
    <li>python311Packages.djangorestframework-dataclasses.dist</li>
    <li>python311Packages.djangorestframework-guardian2</li>
    <li>python311Packages.djangorestframework-guardian2.dist</li>
    <li>python311Packages.djangorestframework-recursive</li>
    <li>python311Packages.djangorestframework-recursive.dist</li>
    <li>python311Packages.djangorestframework-simplejwt</li>
    <li>python311Packages.djangorestframework-simplejwt.dist</li>
    <li>python311Packages.djangorestframework.dist</li>
    <li>python311Packages.drf-jwt</li>
    <li>python311Packages.drf-jwt.dist</li>
    <li>python311Packages.drf-nested-routers</li>
    <li>python311Packages.drf-nested-routers.dist</li>
    <li>python311Packages.drf-spectacular</li>
    <li>python311Packages.drf-spectacular.dist</li>
    <li>python311Packages.drf-ujson2</li>
    <li>python311Packages.drf-ujson2.dist</li>
    <li>python311Packages.drf-writable-nested</li>
    <li>python311Packages.drf-writable-nested.dist</li>
    <li>python311Packages.drf-yasg</li>
    <li>python311Packages.drf-yasg.dist</li>
    <li>python311Packages.graphene-django</li>
    <li>python311Packages.graphene-django.dist</li>
    <li>python311Packages.icontract</li>
    <li>python311Packages.icontract.dist</li>
    <li>python311Packages.permissionedforms</li>
    <li>python311Packages.permissionedforms.dist</li>
    <li>python311Packages.pylddwrap</li>
    <li>python311Packages.pylddwrap.dist</li>
    <li>python311Packages.rules</li>
    <li>python311Packages.rules.dist</li>
    <li>python311Packages.vaa</li>
    <li>python311Packages.vaa.dist</li>
    <li>python311Packages.wagtail</li>
    <li>python311Packages.wagtail-factories</li>
    <li>python311Packages.wagtail-factories.dist</li>
    <li>python311Packages.wagtail-localize</li>
    <li>python311Packages.wagtail-localize.dist</li>
    <li>python311Packages.wagtail-modeladmin</li>
    <li>python311Packages.wagtail-modeladmin.dist</li>
    <li>python311Packages.wagtail.dist</li>
    <li>python312Packages.deal</li>
    <li>python312Packages.deal.dist</li>
    <li>python312Packages.dj-rest-auth</li>
    <li>python312Packages.dj-rest-auth.dist</li>
    <li>python312Packages.django-celery-beat</li>
    <li>python312Packages.django-celery-beat.dist</li>
    <li>python312Packages.django-filter</li>
    <li>python312Packages.django-filter.dist</li>
    <li>python312Packages.django-i18nfield</li>
    <li>python312Packages.django-i18nfield.dist</li>
    <li>python312Packages.django-login-required-middleware</li>
    <li>python312Packages.django-login-required-middleware.dist</li>
    <li>python312Packages.django-modelcluster</li>
    <li>python312Packages.django-modelcluster.dist</li>
    <li>python312Packages.django-oauth-toolkit</li>
    <li>python312Packages.django-oauth-toolkit.dist</li>
    <li>python312Packages.django-payments</li>
    <li>python312Packages.django-payments.dist</li>
    <li>python312Packages.django-phonenumber-field</li>
    <li>python312Packages.django-phonenumber-field.dist</li>
    <li>python312Packages.django-postgresql-netfields</li>
    <li>python312Packages.django-postgresql-netfields.dist</li>
    <li>python312Packages.django-rest-auth</li>
    <li>python312Packages.django-rest-auth.dist</li>
    <li>python312Packages.django-rest-polymorphic</li>
    <li>python312Packages.django-rest-polymorphic.dist</li>
    <li>python312Packages.django-rest-registration</li>
    <li>python312Packages.django-rest-registration.dist</li>
    <li>python312Packages.django-taggit</li>
    <li>python312Packages.django-taggit.dist</li>
    <li>python312Packages.django-timezone-field</li>
    <li>python312Packages.django-timezone-field.dist</li>
    <li>python312Packages.django-two-factor-auth</li>
    <li>python312Packages.django-two-factor-auth.dist</li>
    <li>python312Packages.djangorestframework</li>
    <li>python312Packages.djangorestframework-camel-case</li>
    <li>python312Packages.djangorestframework-camel-case.dist</li>
    <li>python312Packages.djangorestframework-dataclasses</li>
    <li>python312Packages.djangorestframework-dataclasses.dist</li>
    <li>python312Packages.djangorestframework-guardian2</li>
    <li>python312Packages.djangorestframework-guardian2.dist</li>
    <li>python312Packages.djangorestframework-recursive</li>
    <li>python312Packages.djangorestframework-recursive.dist</li>
    <li>python312Packages.djangorestframework-simplejwt</li>
    <li>python312Packages.djangorestframework-simplejwt.dist</li>
    <li>python312Packages.djangorestframework.dist</li>
    <li>python312Packages.drf-jwt</li>
    <li>python312Packages.drf-jwt.dist</li>
    <li>python312Packages.drf-nested-routers</li>
    <li>python312Packages.drf-nested-routers.dist</li>
    <li>python312Packages.drf-spectacular</li>
    <li>python312Packages.drf-spectacular.dist</li>
    <li>python312Packages.drf-ujson2</li>
    <li>python312Packages.drf-ujson2.dist</li>
    <li>python312Packages.drf-writable-nested</li>
    <li>python312Packages.drf-writable-nested.dist</li>
    <li>python312Packages.drf-yasg</li>
    <li>python312Packages.drf-yasg.dist</li>
    <li>python312Packages.icontract</li>
    <li>python312Packages.icontract.dist</li>
    <li>python312Packages.permissionedforms</li>
    <li>python312Packages.permissionedforms.dist</li>
    <li>python312Packages.pylddwrap</li>
    <li>python312Packages.pylddwrap.dist</li>
    <li>python312Packages.rules</li>
    <li>python312Packages.rules.dist</li>
    <li>python312Packages.vaa</li>
    <li>python312Packages.vaa.dist</li>
    <li>python312Packages.wagtail</li>
    <li>python312Packages.wagtail-factories</li>
    <li>python312Packages.wagtail-factories.dist</li>
    <li>python312Packages.wagtail-localize</li>
    <li>python312Packages.wagtail-localize.dist</li>
    <li>python312Packages.wagtail-modeladmin</li>
    <li>python312Packages.wagtail-modeladmin.dist</li>
    <li>python312Packages.wagtail.dist</li>
    <li>tandoor-recipes</li>
  </ul>
</details>

**Comment by matthiasbeyer** [comment]
django-dynamic-preferences maintainer @mmai - what about the build failure here? Do we care?

**Comment by sephii** [comment]
django-dynamic-preferences build fails with Python 3.12 because its tests require distutils which has been removed in Python 3.12.

The failure is not related to this PR so I don’t think it should block the merge. If we’d like to keep the package available on Python 3.12 I suggest we just disable its tests on this specific version until it’s fixed upstream:

```nix
{
  pythonImportsCheck = [ "dynamic_preferences" ];
  doCheck = pythonOlder "3.12";
}
```

**Comment by mweinelt** [comment]
Result of `nixpkgs-review pr 315187` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages marked as broken and skipped:</summary>
  <ul>
    <li>python311Packages.djangorestframework-guardian</li>
    <li>python311Packages.djangorestframework-guardian.dist</li>
    <li>python312Packages.django-countries</li>
    <li>python312Packages.django-countries.dist</li>
    <li>python312Packages.django-graphiql-debug-toolbar</li>
    <li>python312Packages.django-graphiql-debug-toolbar.dist</li>
    <li>python312Packages.djangorestframework-guardian</li>
    <li>python312Packages.djangorestframework-guardian.dist</li>
    <li>python312Packages.graphene-django</li>
    <li>python312Packages.graphene-django.dist</li>
  </ul>
</details>
<details>
  <summary>5 packages failed to build:</summary>
  <ul>
    <li>authentik</li>
    <li>paperless-ngx</li>
    <li>pretalx</li>
    <li>pretalx.dist</li>
    <li>pretalx.static</li>
  </ul>
</details>
<details>
  <summary>169 packages built:</summary>
  <ul>
    <li>mailman-web</li>
    <li>mailman-web.dist</li>
    <li>netbox (netbox_3_7)</li>
    <li>peering-manager</li>
    <li>pretix</li>
    <li>pretix.dist</li>
    <li>python311Packages.deal</li>
    <li>python311Packages.deal.dist</li>
    <li>python311Packages.dj-rest-auth</li>
    <li>python311Packages.dj-rest-auth.dist</li>
    <li>python311Packages.django-celery-beat</li>
    <li>python311Packages.django-celery-beat.dist</li>
    <li>python311Packages.django-countries</li>
    <li>python311Packages.django-countries.dist</li>
    <li>python311Packages.django-dynamic-preferences</li>
    <li>python311Packages.django-dynamic-preferences.dist</li>
    <li>python311Packages.django-filter</li>
    <li>python311Packages.django-filter.dist</li>
    <li>python311Packages.django-graphiql-debug-toolbar</li>
    <li>python311Packages.django-graphiql-debug-toolbar.dist</li>
    <li>python311Packages.django-i18nfield</li>
    <li>python311Packages.django-i18nfield.dist</li>
    <li>python311Packages.django-login-required-middleware</li>
    <li>python311Packages.django-login-required-middleware.dist</li>
    <li>python311Packages.django-modelcluster</li>
    <li>python311Packages.django-modelcluster.dist</li>
    <li>python311Packages.django-oauth-toolkit</li>
    <li>python311Packages.django-oauth-toolkit.dist</li>
    <li>python311Packages.django-payments</li>
    <li>python311Packages.django-payments.dist</li>
    <li>python311Packages.django-phonenumber-field</li>
    <li>python311Packages.django-phonenumber-field.dist</li>
    <li>python311Packages.django-postgresql-netfields</li>
    <li>python311Packages.django-postgresql-netfields.dist</li>
    <li>python311Packages.django-rest-auth</li>
    <li>python311Packages.django-rest-auth.dist</li>
    <li>python311Packages.django-rest-polymorphic</li>
    <li>python311Packages.django-rest-polymorphic.dist</li>
    <li>python311Packages.django-rest-registration</li>
    <li>python311Packages.django-rest-registration.dist</li>
    <li>python311Packages.django-taggit</li>
    <li>python311Packages.django-taggit.dist</li>
    <li>python311Packages.django-timezone-field</li>
    <li>python311Packages.django-timezone-field.dist</li>
    <li>python311Packages.django-two-factor-auth</li>
    <li>python311Packages.django-two-factor-auth.dist</li>
    <li>python311Packages.djangorestframework</li>
    <li>python311Packages.djangorestframework-camel-case</li>
    <li>python311Packages.djangorestframework-camel-case.dist</li>
    <li>python311Packages.djangorestframework-dataclasses</li>
    <li>python311Packages.djangorestframework-dataclasses.dist</li>
    <li>python311Packages.djangorestframework-guardian2</li>
    <li>python311Packages.djangorestframework-guardian2.dist</li>
    <li>python311Packages.djangorestframework-recursive</li>
    <li>python311Packages.djangorestframework-recursive.dist</li>
    <li>python311Packages.djangorestframework-simplejwt</li>
    <li>python311Packages.djangorestframework-simplejwt.dist</li>
    <li>python311Packages.djangorestframework.dist</li>
    <li>python311Packages.drf-jwt</li>
    <li>python311Packages.drf-jwt.dist</li>
    <li>python311Packages.drf-nested-routers</li>
    <li>python311Packages.drf-nested-routers.dist</li>
    <li>python311Packages.drf-spectacular</li>
    <li>python311Packages.drf-spectacular.dist</li>
    <li>python311Packages.drf-ujson2</li>
    <li>python311Packages.drf-ujson2.dist</li>
    <li>python311Packages.drf-writable-nested</li>
    <li>python311Packages.drf-writable-nested.dist</li>
    <li>python311Packages.drf-yasg</li>
    <li>python311Packages.drf-yasg.dist</li>
    <li>python311Packages.graphene-django</li>
    <li>python311Packages.graphene-django.dist</li>
    <li>python311Packages.icontract</li>
    <li>python311Packages.icontract.dist</li>
    <li>python311Packages.permissionedforms</li>
    <li>python311Packages.permissionedforms.dist</li>
    <li>python311Packages.pylddwrap</li>
    <li>python311Packages.pylddwrap.dist</li>
    <li>python311Packages.rules</li>
    <li>python311Packages.rules.dist</li>
    <li>python311Packages.vaa</li>
    <li>python311Packages.vaa.dist</li>
    <li>python311Packages.wagtail</li>
    <li>python311Packages.wagtail-factories</li>
    <li>python311Packages.wagtail-factories.dist</li>
    <li>python311Packages.wagtail-localize</li>
    <li>python311Packages.wagtail-localize.dist</li>
    <li>python311Packages.wagtail-modeladmin</li>
    <li>python311Packages.wagtail-modeladmin.dist</li>
    <li>python311Packages.wagtail.dist</li>
    <li>python312Packages.deal</li>
    <li>python312Packages.deal.dist</li>
    <li>python312Packages.dj-rest-auth</li>
    <li>python312Packages.dj-rest-auth.dist</li>
    <li>python312Packages.django-celery-beat</li>
    <li>python312Packages.django-celery-beat.dist</li>
    <li>python312Packages.django-dynamic-preferences</li>
    <li>python312Packages.django-dynamic-preferences.dist</li>
    <li>python312Packages.django-filter</li>
    <li>python312Packages.django-filter.dist</li>
    <li>python312Packages.django-i18nfield</li>
    <li>python312Packages.django-i18nfield.dist</li>
    <li>python312Packages.django-login-required-middleware</li>
    <li>python312Packages.django-login-required-middleware.dist</li>
    <li>python312Packages.django-modelcluster</li>
    <li>python312Packages.django-modelcluster.dist</li>
    <li>python312Packages.django-oauth-toolkit</li>
    <li>python312Packages.django-oauth-toolkit.dist</li>
    <li>python312Packages.django-payments</li>
    <li>python312Packages.django-payments.dist</li>
    <li>python312Packages.django-phonenumber-field</li>
    <li>python312Packages.django-phonenumber-field.dist</li>
    <li>python312Packages.django-postgresql-netfields</li>
    <li>python312Packages.django-postgresql-netfields.dist</li>
    <li>python312Packages.django-rest-auth</li>
    <li>python312Packages.django-rest-auth.dist</li>
    <li>python312Packages.django-rest-polymorphic</li>
    <li>python312Packages.django-rest-polymorphic.dist</li>
    <li>python312Packages.django-rest-registration</li>
    <li>python312Packages.django-rest-registration.dist</li>
    <li>python312Packages.django-taggit</li>
    <li>python312Packages.django-taggit.dist</li>
    <li>python312Packages.django-timezone-field</li>
    <li>python312Packages.django-timezone-field.dist</li>
    <li>python312Packages.django-two-factor-auth</li>
    <li>python312Packages.django-two-factor-auth.dist</li>
    <li>python312Packages.djangorestframework</li>
    <li>python312Packages.djangorestframework-camel-case</li>
    <li>python312Packages.djangorestframework-camel-case.dist</li>
    <li>python312Packages.djangorestframework-dataclasses</li>
    <li>python312Packages.djangorestframework-dataclasses.dist</li>
    <li>python312Packages.djangorestframework-guardian2</li>
    <li>python312Packages.djangorestframework-guardian2.dist</li>
    <li>python312Packages.djangorestframework-recursive</li>
    <li>python312Packages.djangorestframework-recursive.dist</li>
    <li>python312Packages.djangorestframework-simplejwt</li>
    <li>python312Packages.djangorestframework-simplejwt.dist</li>
    <li>python312Packages.djangorestframework.dist</li>
    <li>python312Packages.drf-jwt</li>
    <li>python312Packages.drf-jwt.dist</li>
    <li>python312Packages.drf-nested-routers</li>
    <li>python312Packages.drf-nested-routers.dist</li>
    <li>python312Packages.drf-spectacular</li>
    <li>python312Packages.drf-spectacular.dist</li>
    <li>python312Packages.drf-ujson2</li>
    <li>python312Packages.drf-ujson2.dist</li>
    <li>python312Packages.drf-writable-nested</li>
    <li>python312Packages.drf-writable-nested.dist</li>
    <li>python312Packages.drf-yasg</li>
    <li>python312Packages.drf-yasg.dist</li>
    <li>python312Packages.icontract</li>
    <li>python312Packages.icontract.dist</li>
    <li>python312Packages.permissionedforms</li>
    <li>python312Packages.permissionedforms.dist</li>
    <li>python312Packages.pylddwrap</li>
    <li>python312Packages.pylddwrap.dist</li>
    <li>python312Packages.rules</li>
    <li>python312Packages.rules.dist</li>
    <li>python312Packages.vaa</li>
    <li>python312Packages.vaa.dist</li>
    <li>python312Packages.wagtail</li>
    <li>python312Packages.wagtail-factories</li>
    <li>python312Packages.wagtail-factories.dist</li>
    <li>python312Packages.wagtail-localize</li>
    <li>python312Packages.wagtail-localize.dist</li>
    <li>python312Packages.wagtail-modeladmin</li>
    <li>python312Packages.wagtail-modeladmin.dist</li>
    <li>python312Packages.wagtail.dist</li>
    <li>tandoor-recipes</li>
  </ul>
</details>

**Pull request merged**
